llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -272,13 +272,15 @@ RISCVTargetLowering::RISCVTargetLowering(const TargetMachine &TM,
   setOperationAction({ISD::SHL_PARTS, ISD::SRL_PARTS, ISD::SRA_PARTS}, XLenVT,
                      Custom);
 
-  if (Subtarget.hasStdExtZbb() || Subtarget.hasStdExtZbkb()) {
-    if (Subtarget.is64Bit())
-      setOperationAction({ISD::ROTL, ISD::ROTR}, MVT::i32, Custom);
-  } else {
-    setOperationAction({ISD::ROTL, ISD::ROTR}, XLenVT, Expand);
+  
+  if (!Subtarget.hasExtXcvbitmanip()) {
```

Comment:
@ PaoloS02 This is something you might want to have a look at.
I have edited this code because RISCV Base ISA doesn't have a rotate right instruction and thus during legalization, it expands the ROTR operation into a combination of other supported operations (srl, neg, sll ,or).